dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EventResource.java

```diff
@@ -84,18 +85,36 @@ public EventResource(EventDAO eventDAO, UserDAO userDAO) {
         switch (eventSearchType) {
         case STARRED_ENTRIES:
             Set<Long> entryIDs = userWithSession.getStarredEntries().stream().map(Entry::getId).collect(Collectors.toSet());
-            return this.eventDAO.findEventsByEntryIDs(entryIDs, offset, limit);
+            List<Event> eventsByEntryIDs = this.eventDAO.findEventsByEntryIDs(entryIDs, offset, limit);
+            eagerLoadEventEntries(eventsByEntryIDs);
+            return eventsByEntryIDs;
         case STARRED_ORGANIZATION:
             Set<Long> organizationIDs = userWithSession.getStarredOrganizations().stream().map(Organization::getId).collect(Collectors.toSet());
-            return this.eventDAO.findAllByOrganizationIds(organizationIDs, offset, limit);
+            List<Event> allByOrganizationIds = this.eventDAO.findAllByOrganizationIds(organizationIDs, offset, limit);
+            eagerLoadEventEntries(allByOrganizationIds);
+            return allByOrganizationIds;
         case ALL_STARRED:
             Set<Long> organizationIDs2 = userWithSession.getStarredOrganizations().stream().map(Organization::getId).collect(Collectors.toSet());
             Set<Long> entryIDs2 = userWithSession.getStarredEntries().stream().map(Entry::getId).collect(Collectors.toSet());
-            return this.eventDAO.findAllByOrganizationIdsOrEntryIds(organizationIDs2, entryIDs2, offset, limit);
+            List<Event> allByOrganizationIdsOrEntryIds = this.eventDAO
+                    .findAllByOrganizationIdsOrEntryIds(organizationIDs2, entryIDs2, offset, limit);
+            eagerLoadEventEntries(allByOrganizationIdsOrEntryIds);
+            return allByOrganizationIdsOrEntryIds;
         default:
             return Collections.emptyList();
         }
     }
+
+    private void eagerLoadEventEntries(List<Event> events) {
+        events.forEach(event -> {
```

Comment:
Do each of these create a separate db connection?

Reply:
No, same amount of connections acquired and released without it